test/unit/getPastEventsFromHub.test.ts

```diff
@@ -0,0 +1,395 @@
+      config.util.extendDeep(config, { blockchain: { maxBlockRange: 10 } });
+      const relayHubAddress = '0x456efc';
+      const expectedEncodedAddress = 'encodedAddress';
+      const stubRelayHub = {
+        interface: {
+          getEventTopic: (name: string) => `encodedTopic-${name}`,
+          _abiCoder: {
+            encode: () => expectedEncodedAddress,
+          },
+        },
+        address: relayHubAddress,
+      } as unknown as RelayHub;
+      const expectedEncodedTopic = [
+        'encodedTopic-RelayServerRegistered',
+        'encodedTopic-RelayWorkersAdded',
+        'encodedTopic-TransactionRelayed',
+      ];
+      const managerAddress = '0x123abc';
+      const logFilters = await getLogFilters(
+        managerAddress,
+        { fromBlock: 1, toBlock: 10 },
+        ['RelayServerRegistered', 'RelayWorkersAdded', 'TransactionRelayed'],
+        stubRelayHub
+      );
+
+      const expectedLogFilters = [
+        {
+          address: '0x456efc',
+          topics: [expectedEncodedTopic, [expectedEncodedAddress]],
+          fromBlock: 1,
+          toBlock: 10,
+        },
+      ];
+
+      expect(logFilters).to.be.eql(expectedLogFilters);
+    });
+  });
+
+  describe('performLogRequests', function () {
+    const getLogsResponse = [
+      {
+        blockNumber: 1,
+        blockHash: 'hash',
+        transactionIndex: 1,
+        removed: false,
+        address: 'aaa',
+        data: 'bbb',
+        topics: ['topicA', 'topicB'],
+        transactionHash: 'txHash',
+        logIndex: '0',
+      },
+    ];
+
+    it('should return the logs if it involves one request only', async function () {
+      const stubProvider = {
+        getLogs: () => getLogsResponse,
+      } as unknown as providers.Provider;
+      const logs = await performLogRequests(
+        [
+          {
+            fromBlock: 1,
+            toBlock: 10,
+            address: '0x123abc',
+            topics: [['eventA'], ['managerAddress']],
+          },
+        ],
+        stubProvider
+      );
+      expect(logs).to.be.eql(getLogsResponse);
+    });
+
+    it('should return the logs if it involves multiple requests', async function () {
+      const stubProvider = {
+        getLogs: () => getLogsResponse,
+      } as unknown as providers.Provider;
+      const logs = await performLogRequests(
+        [
+          {
+            fromBlock: 1,
+            toBlock: 11,
+            address: '0x123abc',
+            topics: [['eventA'], ['managerAddress']],
+          },
+          {
+            fromBlock: 11,
+            toBlock: 21,
+            address: '0x123abc',
+            topics: [['eventA'], ['managerAddress']],
+          },
+          {
+            fromBlock: 21,
+            toBlock: 31,
+            address: '0x123abc',
+            topics: [['eventA'], ['managerAddress']],
+          },
+        ],
+        stubProvider
+      );
+      const expectedLogs = [
+        ...getLogsResponse,
```

Comment:
Maybe Im being to picky but  the expected logs are always for blockNumber: 1, but we are sending different block ranges.